backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/utils/ApplicationHandler.kt

```diff
@@ -0,0 +1,110 @@
+    private val application: Application,
+    private val regionId: Int,
+    private val project: String
+) {
+
+    fun sendApplicationMails(
+        applicationEntity: ApplicationEntity,
+        verificationEntities: List<ApplicationVerificationEntity>,
+        dataFetcherResultBuilder: DataFetcherResult.Builder<Boolean>
+    ) {
+        val backendConfig = context.backendConfiguration
+        val projectConfig = backendConfig.projects.first { it.id == project }
+
+        Mailer.sendApplicationApplicantMail(
+            backendConfig,
+            projectConfig,
+            application.personalData,
+            applicationEntity.accessKey
+        )
+
+        for (applicationVerification in verificationEntities) {
+            try {
+                val applicantName =
+                    "${application.personalData.forenames.shortText} ${application.personalData.surname.shortText}"
+                Mailer.sendApplicationVerificationMail(
+                    backendConfig,
+                    applicantName,
+                    projectConfig,
+                    applicationVerification
+                )
+            } catch (exception: MailNotSentException) {
+                dataFetcherResultBuilder.error(exception.toError())
+            }
+        }
+        Mailer.sendNotificationForApplicationMails(project, backendConfig, projectConfig, regionId)
+    }
+
+    fun validateAttachmentTypes() {
+        val allowedContentTypes = setOf("application/pdf", "image/png", "image/jpeg")
+        val maxFileSizeBytes = 5 * 1000 * 1000
+        if (!context.files.all { it.contentType in allowedContentTypes }) {
+            throw InvalidFileTypeException()
+        }
+        if (!context.files.all { it.size <= maxFileSizeBytes }) {
+            throw InvalidFileSizeException()
+        }
+    }
+
+    fun validateRegion() {
+        val region = transaction { RegionsRepository.findByIdInProject(project, regionId) }
+            ?: throw RegionNotFoundException()
+        if (!region.activatedForApplication) {
+            throw RegionNotActivatedForApplicationException()
+        }
+    }
+
+    fun saveApplication(): Pair<ApplicationEntity, List<ApplicationVerificationEntity>> {
+        val (applicationEntity, verificationEntities) = transaction {
+            ApplicationRepository.persistApplication(
+                application.toJsonField(),
+                application.extractApplicationVerifications(),
+                regionId,
+                context.applicationData,
+                context.files
+            )
+        }
+        return Pair(applicationEntity, verificationEntities)
+    }
+
+    fun isValidPreVerifiedApplication(): Boolean {
+        val isAlreadyVerifiedIsSet =
+            application.applicationDetails.blueCardEntitlement?.workAtOrganizationsEntitlement?.list?.any {
+                it.isAlreadyVerified == true
+            } ?: false
+        if (isAlreadyVerifiedIsSet) {
```

Comment:
```suggestion
        if (isAlreadyVerifiedSet) {
```
:upside_down_face:
i think "is" is duplicated here. Sounds bit strange to me :)